master/internal/hpimportance/hpimportance.go

```diff
@@ -1,15 +1,227 @@
+	minNumberTrials  = 50
+	idealBatchDiff   = 2
+	nReplications    = 2
+
+	arffFile       = "data.arff"
+	importanceFile = "importance.txt"
+)
+
+// The data needs to be put into an arff format.
+// Arff files require variable declaration at the top,
+// where order does matter. We need to keep define the Hps and
+// keep track of the order so the data columns will match.
+func createDataFile(data map[int][]model.HPImportanceTrialData,
+	experimentConfig *model.ExperimentConfig, workingDir string) (int, error) {
+	f, err := os.Create(path.Join(workingDir, arffFile))
+	if err != nil {
+		return 0, err
+	}
+	defer func() {
+		err = f.Close()
+		if err != nil {
+			fmt.Printf("failed to close arff file: %s", err.Error())
+		}
+	}()
+
+	// create top of file based on exp config
+	_, err = f.WriteString("@relation data\n\n@attribute metric numeric\n")
+	if err != nil {
+		return 0, err
+	}
+	var hpsOrder []string // HPs must be in the same order for the arff file
+	hps := experimentConfig.Hyperparameters
+
+	for key, element := range hps {
+		var st string
+		switch {
+		case element.ConstHyperparameter != nil:
+			continue
+		case element.CategoricalHyperparameter != nil:
+			hpsOrder = append(hpsOrder, key)
+
+			var values string
+			for _, catVal := range element.CategoricalHyperparameter.Vals {
+				values += fmt.Sprintf("%v,", catVal)
+			}
+			st = fmt.Sprintf("@attribute %s {%s}\n", key, values)
+		default:
+			hpsOrder = append(hpsOrder, key)
+			st = fmt.Sprintf("@attribute %s numeric\n", key)
+		}
+		_, err = f.WriteString(st)
+		if err != nil {
+			return 0, err
+		}
+	}
+	_, err = f.WriteString("@attribute numBatches numeric\n\n")
+	if err != nil {
+		return 0, err
+	}
+
+	// Now we have to add the data to the file
+	_, err = f.WriteString("\n@data\n")
+	if err != nil {
+		return 0, err
+	}
+
+	// First, get batch ids unless better way to get them in go?
+	var batches []int
+	for k := range data {
+		batches = append(batches, k)
+	}
+	sort.Sort(sort.Reverse(sort.IntSlice(batches)))
+
+	totalNumTrials := 0
+	maxNumBatches := batches[0]
+	for _, batchID := range batches {
+		if batchID < maxNumBatches/maxDiffCompBatch {
+			break
+		}
+		if totalNumTrials > minNumberTrials && batchID <= maxNumBatches/idealBatchDiff {
+			break
+		}
+		batchIDStr := fmt.Sprintf("%v", batchID)
+		for _, trial := range data[batchID] {
+			var st string
+			st += fmt.Sprintf("%v", trial.Metric)
+			hparamsVals := trial.Hparams
+			for _, hp := range hpsOrder {
+				st += fmt.Sprintf(",%v", hparamsVals[hp])
+			}
+			st += fmt.Sprintf(",%s\n", batchIDStr)
+			_, err = f.WriteString(st)
+			if err != nil {
+				return 0, err
+			}
+
+			totalNumTrials++
+		}
 	}
-	return output
+	return totalNumTrials, nil
+}
+
+// Read the data from the output importance file and return as a json.
+func parseImportanceOutput(filename string) (map[string]float64, error) {
+	// Ignore security warning because none of this is user-provided input
+	// #nosec G304
```

Comment:
```suggestion
	// #nosec G304 // Ignore security warning because none of this is user-provided input
```
usually the explanation goes like this